libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Memory/DialogStateManager.cs

```diff
@@ -464,6 +514,58 @@ private string GetBadScopeMessage(string path)
             return $"'{path}' does not match memory scopes:{string.Join(",", Configuration.MemoryScopes.Select(ms => ms.Name))}";
         }
 
+        private bool TrackChange(string path, object value)
```

Comment:
>TrackChange [](start = 21, length = 11)

can you describe what this is doing?  I don't understand all of the "_" logic in here